File: free-platform-book/free-platform-book-api/src/main/java/top/yangwulang/platform/entity/book/BookCategory.java
package top.yangwulang.platform.entity.book;

import io.swagger.v3.oas.annotations.media.Schema;
import org.babyfish.jimmer.Immutable;
import org.babyfish.jimmer.sql.*;

import javax.validation.constraints.NotNull;
import javax.validation.constraints.Null;

/**
 * Entity for table "book_category"
 */
@Entity
@Table(name = "book_category")
@Schema(name = "书籍分类")
public interface BookCategory {
    /**
     * 书籍分类id
     */
    @Id
    String id();

    /**
     * 分类名称
     */
    @Null String categoryName();
}

File: platform/src/main/kotlin/top/yangwulang/platform/entity/book/BookCategory.kt
package top.yangwulang.platform.entity.book

import org.hibernate.annotations.GenericGenerator
import java.io.Serializable
import javax.persistence.*

@Entity
@Table(name = "book_category")
open class BookCategory : Serializable {
    @Id
    @Column(name = "id", length = 64)
    @GeneratedValue(generator = "snowflake")
    @GenericGenerator(name = "snowflake", strategy = "top.yangwulang.platform.utils.SnowflakeGenerator")
    open var id: String? = null

    @Column(name = "category_name", length = 500)
    open var categoryName: String? = null
}